Standalone version isn't loaded with ESM and Webpack

**Environments:**

- Prettier Version: 3.0 alpha
- Running Prettier via: Browser API
- Prettier plugins (if any): <!-- e.g. @prettier/plugin-php -->

**Steps to reproduce:**

- import `prettier` in ESM-only React code (`"type": "module"`, and Typescript in my case, but I'm not sure that's relevant)
- try to bundle it with webpack 5

**Expected behavior:**

Bundle is built.

**Actual behavior:**

Multiple errors of this type:

```
ERROR in ../../node_modules/prettier/index.mjs 1:0-66
Module not found: Error: Can't resolve 'module' in '/Users/berekuk/coding/quri/squiggle/node_modules/prettier'
 @ ./src/components/CodeEditor.tsx 11:0-37 41:50-75
 @ ./src/components/SquiggleEditor.tsx 5:0-45 20:32-42
 @ ./src/index.ts 4:0-64 4:0-64
```

This could be solved with `prettier: "prettier/standalone"` in webpack's `resolve.alias` field, but that's a solution only for the current package; I have multiple apps that use the same components package, and I don't want to impose this requirement on all consumers of my package.

Importing `prettier/standalone` directly in the React code doesn't help either because I depend on the custom prettier plugin that imports `prettier` instead of `prettier/standalone`.

I think that adding `"browser": "./standalone.mjs"` to `exports["."]` field in `package.json` will fix this. I'm less sure if it would cause any problems for CJS builds.

--- a/scripts/build/build-package-json.js
+++ b/scripts/build/build-package-json.js
@@ -42,6 +42,10 @@ async function buildPackageJson({ file, files }) {
       ".": {
         types: "./index.d.ts",
         require: "./index.cjs",
+        browser: {
+          import: "./standalone.mjs",
+          default: "./standalone.js",
+        },
         default: "./index.mjs",
       },
       "./*": "./*",

